pkg/network/vmispec/hotplug.go

```diff
@@ -32,10 +36,39 @@ func NetworksToHotplug(vmi *v1.VirtualMachineInstance) []v1.Network {
 	return networksToHotplug
 }
 
-func indexedInterfacesFromStatus(interfaces []v1.VirtualMachineInstanceNetworkInterface) map[string]v1.VirtualMachineInstanceNetworkInterface {
+func indexedInterfacesFromStatus(interfaces []v1.VirtualMachineInstanceNetworkInterface, predicates ...func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool) map[string]v1.VirtualMachineInstanceNetworkInterface {
 	indexedInterfaceStatus := map[string]v1.VirtualMachineInstanceNetworkInterface{}
 	for _, iface := range interfaces {
-		indexedInterfaceStatus[iface.Name] = iface
+		if len(predicates) == 0 {
+			indexedInterfaceStatus[iface.Name] = iface
+			continue
+		}
+		for _, p := range predicates {
+			if p(iface) {
+				indexedInterfaceStatus[iface.Name] = iface
+				break
+			}
+		}
 	}
 	return indexedInterfaceStatus
 }
+
+func NetworksToHotplugWhosePodIfacesAreReady(vmi *v1.VirtualMachineInstance) []v1.Network {
+	var networksToHotplug []v1.Network
+	interfacesToHoplug := indexedInterfacesFromStatus(
+		vmi.Status.Interfaces,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
+			return ifaceStatus.PodConfigDone && !strings.Contains(
+				ifaceStatus.InfoSource, InfoSourceDomain,
```

Comment:
It doesn't - when we compute the interface status, all the interfaces that appear listed in the active domain will feature this source.

And we compute the interface status after the `SyncVMI` call.

Reply:
virt-launcher has to get the interfaces status from libvirt and then report it to virt-handler which in turn will update the status. I still don't understand what blocks virt-handler from trying to hotplug the interface again and again until the status is updated.

Reply:
OK @AlonaKaplan , I now understand what you meant.

virt-handler will indeed attempt to plug again, but it'll find the networking infrastructure (bridge / tap) already setup in the pod - due to the `PodIfaceNetworkPreparationFinished` [flag](https://github.com/kubevirt/kubevirt/blob/603827c091e741774c13c87d968a754a12400d55/pkg/network/setup/configstate.go#L59) and safely exit.

... this is not exactly for free, since this happens when the handler is tapping into the launcher pod's net namespace, but I think it's assumable since the same thing happens when creating an interface.

Reply:
When running a new vm you won't enter the launcher network namespace more than once because of the [setupCompleted](https://github.com/kubevirt/kubevirt/blob/25b37338917f1a7bfbe3fc07d87672ef1e39389c/pkg/network/setup/netconf.go#L69) protection.

But anyway, I think we can live with it on hotplug. The comment can be resolved.